Makefile

```diff
@@ -0,0 +1,73 @@
+# By default do a release build with moving immix
+MMTK_MOVING ?= 1
+MMTK_PLAN ?= Immix
+CURR_PATH := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+
+# Getting metadata about Julia repo from Cargo file
+JULIA_GIT_URL= $(shell cargo read-manifest --manifest-path=$(CURR_PATH)/mmtk/Cargo.toml | python -c 'import json,sys; print(json.load(sys.stdin)["metadata"]["julia"]["julia_repo"])')
+JULIA_VERSION= $(shell cargo read-manifest --manifest-path=$(CURR_PATH)/mmtk/Cargo.toml | python -c 'import json,sys; print(json.load(sys.stdin)["metadata"]["julia"]["julia_version"])')
+
+# If the Julia directory doesn't exist
+# Clone it as a sibling of mmtk-julia
+JULIA_PATH ?= $(CURR_PATH)../julia
+MMTK_JULIA_DIR := $(CURR_PATH)
+
+PROJECT_DIRS := JULIA_PATH=$(JULIA_PATH) MMTK_JULIA_DIR=$(MMTK_JULIA_DIR)
+MMTK_VARS := MMTK_PLAN=$(MMTK_PLAN) MMTK_MOVING=$(MMTK_MOVING)
+
+ifeq (${MMTK_PLAN},Immix)
+CARGO_FEATURES = immix
+else ifeq (${MMTK_PLAN},StickyImmix)
+CARGO_FEATURES = stickyimmix
+else
+$(error "Unsupported MMTk plan: $(MMTK_PLAN)")
+endif
+
+ifeq ($(MMTK_MOVING), 0)
+CARGO_FEATURES := $(CARGO_FEATURES),non_moving
+endif
+
+# Clone the repository if the directory does not exist
+clone-julia:
```

Comment:
Why do we need to clone Julia here? We have a build script in the Julia repo to clone this repo. We shouldn't need to clone the Julia repo in the build script here. Simialr to the version issue, we need to break this circular dependency.

Reply:
I don't think we can remove that as we do need the Julia repo to run `bindgen`. Either we keep that or we need to require that the user set the `JULIA_PATH` (and/or `BUILDDIR`), meaning that they will have cloned it already.

Reply:
We want people to clone Julia, and then build Julia, during which MMTk will be cloned and built. In this process, when we have the MMTk repo, we must have the Julia repo. 

If the above is true, I don't get why we need to clone Julia here. It shouldn't be triggered in the above workflow, right?